repo.go

```diff
@@ -299,10 +309,16 @@ func (r *Repo) ChangePassphrase(keyRole string) error {
 }
 
 func (r *Repo) GenKey(role string) ([]string, error) {
```

Comment:
allowing non-tree makes this very complex. See this discussion: https://github.com/theupdateframework/specification/issues/214#issuecomment-1086228742